docs/nebula-design-concepts.md

```diff
@@ -0,0 +1,216 @@
+may be multiple edges of the same or different types at the same time between the
+start (src)and end vertex (dst). In the preceding example, edges are **serve**
+and **likeness**.
+
+### Properties
+
+Both vertices and edges can have properties, which are effectively key/value pairs. We call them **schema** in Nebula Graph. Like Mysql, Nebula Graph is a strong schema database. The name and data type of the properties are determined before the data is written. Vertex can have one or more properties, and edges can have properties too.
+
+In the preceding example, the schema of **player** has three kind of properties:
+
+- id (vid)
+- Name (string)
+- Age (int)
+
+ the schema of **serve** has two kinds of properties:
+
+ - start_year (int)
+ - end_year (int)
+
+### Tags
+
+In Nebula Graph, the type of vertex is called a tag. A vertex must have at least
+one tag, and Nebula allows you to set multiple tags per vertex.
+
+### Edgetype
+
+Each edge has an edgetype, and each edgetype defines the properties of the edge.
+这是按照我的理解方式写的
+-->
+
+
+### Vertex, type of vertex (tag) and the corresponding vertex properties
+
+In Nebula Graph, the type of vertex is called **tag**. **A vertex must have at least one type (tag)** or multiple types (tag). **Each tag has a corresponding set of properties called schema.**
+
+Consider the above picture as example, there are two types of vertices: **player** and **team**. Vertex player's schema has three properties: id (vid), Name(string), and Age(int); vertex team's schema has two properties: id (vid) and Name(string).
+
+Like Mysql, Nebula Graph is a strong schema database. The property name and data type are determined before the data is written.
+
+### Edge, edgetype and the corresponding edge properties
+
+**Edges in Nebula Graph always have a type, a start (src)and an end vertex (dst), and a direction.** They can be self-referencing/looping but can never be dangling (missing a start or end node). **Each edge has an edgetype, and each edgetype defines the properties of the edge.**
+
+In the preceding example, there are two kinds of edges, one is **like** pointing
+from player to player, the property is likeness (double); the other is **serve**,
+pointing from team to serve, the properties are start_year(int) and end_year(int).
+
+It should be noted that there may be **multiple edges of the same or different types
+at the same time** between the start (src)and end vertex (dst).
+
+
+
+## Graph partition
+
+![image][example2]
+
+Since the number of vertices in a super large-scale relational network can reach
+to dozen billions or trillions, and the number of edges can even be dozen trillions,
+the storage of both will exceed the capacity of a general server. Therefore, a
+method is needed to cut the graph elements and store them on different logical partitions. Nebula Graph conducts **edge splitting** to make partitions, and its default strategy is Hash. The number of partitions is static and cannot be changed.
+
```

Comment:
In applications such as social networks, or road networks, the number of vertices and their connections are extremely large. Partitioning large graphs is important for complexity reduction or parallelization. Nebula Graph adopts Edge Separators to cut a graph into smaller components. And the partition number cannot be changed once it is configured.